packages/test/service-load-test/src/loadTestDataStore.ts

```diff
@@ -87,104 +87,146 @@ class LoadTestDataStoreModel {
             throw new Error("taskmanger not available");
         }
 
-        if(reset) {
-            await LoadTestDataStoreModel.waitForCatchup(runtime);
-            runDir.set(startTimeKey,Date.now());
-            runDir.delete(taskTimeKey);
-            counter.increment(-1 * counter.value);
-        }
-
-        return new LoadTestDataStoreModel(
+        const dataModel =  new LoadTestDataStoreModel(
+            root,
             config,
             runtime,
             taskmanager,
             runDir,
             counter,
         );
+
+        if(reset) {
+            await LoadTestDataStoreModel.waitForCatchup(runtime);
+            runDir.set(startTimeKey,Date.now());
+            runDir.delete(taskTimeKey);
+            counter.increment(-1 * counter.value);
+            const partnerCounter = await dataModel.getPartnerCounter();
+            if(partnerCounter !== undefined && partnerCounter.value > 0) {
+                partnerCounter.increment(-1 * partnerCounter.value);
+            }
+        }
+        return dataModel;
     }
 
     private readonly taskId: string;
+    private readonly partnerId: number;
     private taskStartTime: number =0;
 
     private constructor(
+        private readonly root: ISharedDirectory,
         private readonly config: IRunConfig,
         private readonly runtime: IFluidDataStoreRuntime,
         private readonly taskManager: ITaskManager,
         private readonly dir: IDirectory,
         public readonly counter: ISharedCounter,
     ) {
+        const halfClients = Math.floor(this.config.testConfig.numClients / 2);
         // The runners are paired up and each pair shares a single taskId
-        this.taskId = `op_sender${Math.floor(config.runId / 2)}`;
+        this.taskId = `op_sender${config.runId % halfClients}`;
+        this.partnerId = this.config.runId < halfClients
+            ? this.config.runId + halfClients
+            : this.config.runId % halfClients;
+        const changed = (taskId)=>{
+            if(taskId === this.taskId && this.taskStartTime !== 0) {
+                this.dir.set(taskTimeKey, this.totalTaskTime);
+                this.taskStartTime = 0;
+            }
+        };
+        this.taskManager.on("lost", changed);
+        this.taskManager.on("assigned", changed);
     }
 
     public get startTime(): number {
-        return this.dir.get<number>(startTimeKey) ?? 0;
+        return this.dir.get<number>(startTimeKey) ?? Date.now();
```

Comment:
When does this fallback happen?  If never. better to throw IMO

Reply:
it doesn't really matter. i also want to just get ride of all the current timing code, as it's never been right